src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TargetElementAttribute.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNet.Razor.TagHelpers;
+
+namespace Microsoft.AspNet.Razor.Runtime.TagHelpers
+{
+    /// <summary>
+    /// Provides an <see cref="ITagHelper"/>'s target.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class, AllowMultiple = true, Inherited = false)]
+    public sealed class TargetElementAttribute : Attribute
+    {
+        /// <summary>
+        /// Instantiates a new instance of the <see cref="TargetElementAttribute"/> class with <see cref="Tag"/>
+        /// set to <c>*</c>.
+        /// </summary>
+        /// <remarks>A <c>*</c> <see cref="Tag"/> value results in an <see cref="ITagHelper"/> 
+        /// targeting all HTML elements that have the provided <see cref="Attributes"/>.</remarks>
```

Comment:
nit for a number of comments `... that targets all HTML elements with the required <see cref="Attributes"/>.` (provided is particularly confusing)